### PR TITLE
Support for 3.3

### PR DESCRIPTION
**Pull request by arturolinares**
I also updated the version casing, SAT generates the XML attributes with the first letter in uppercase (!).

### CODE REVIEW

**Comment by eclipxe13** [comment]
Thanks Arturo, I see that you only validate version match, but not add any test.
Could you please try to add some test and check that validation rules are still working?

PS: run `vendor/bin/php-cs-fixer fix -v` & `vendor/bin/phpcbf src/ tests/` on your branch.

**Comment by arturolinares** [comment]
Thanks for letting me know about the tests, for some reason I didn't see them. I refactored the way the reader validates the schema version.

Also, this command is showing me an error, maybe I'm missing a parameter? 
```
$ vendor/bin/php-cs-fixer fix --using-cache=no --dry-run --verbose



  [RuntimeException]
  The "--using-cache" option does not exist.
```

**Comment by eclipxe13** [comment]
@arturolinares Do you have any example of CFDI3.3 ? I check with the sample from http://www.sat.gob.mx/informacion_fiscal/factura_electronica/Paginas/requisitos_factura_electronica_cfdi.aspx but its not 3.3

**Comment by eclipxe13** [comment]
Oh, I see more in your pull and one example. Thanks!. I would like to discuss your approach to this changes. Would you like to chat to avoid double work or discard changes? maybe gitter or other way?

About the php-cs-fixert issue I think is because the recent version of composer install and old version, vould you please edit the line on `composer.json` to `"friendsofphp/php-cs-fixer": "^2.4"`

File: src/CFDIReader/CFDIReader.php [hunk 1]
```diff
@@ -143,4 +153,29 @@ private function populateNode(SimpleXMLElement $source, SimpleXMLElement $destin
...
+
+    public function allowDefaultSchemas()
+    {
+        $this->allowedSchemas = [
+            new SchemaRequirement33,
+            new SchemaRequirement32,
+        ];
+    }
+
+    /**
```
Comment (eclipxe13) [inline src/CFDIReader/CFDIReader.php:166]: I like this function, is a good idea! 👍

**Comment by eclipxe13** [comment]
Have to go home now, some comments:

- I would prefer not to use your approach of SchemaRequirementInterface. Maybe just get the required namespaces after validating the compatible version.
- Your latest run on php-cs-fixer was not using the default config (.php_cs.dist), this happens because the version problem and introduce some unwanted changes.

If you are interested to improve this library it would be great to work together and make a plan using something to communicate faster, I just add the gitter channel for this repo https://gitter.im/eclipxe13/buzoncfdi-cfdireader

**Comment by arturolinares** [comment]
Sorry, had to run home. Let’s chat tomorrow.

--
ALG

De: Carlos C Soto <notifications@github.com>
Responder a: eclipxe13/buzoncfdi-cfdireader <reply@reply.github.com>
Fecha: martes, 8 de agosto de 2017, 22:23
Para: eclipxe13/buzoncfdi-cfdireader <buzoncfdi-cfdireader@noreply.github.com>
CC: Arturo Linares Goenaga <arturo.linares@gmail.com>, Mention <mention@noreply.github.com>
Asunto: Re: [eclipxe13/buzoncfdi-cfdireader] Support for 3.3 (#2)


Have to go home now, some comments:

  *   I would prefer not to use your approach of SchemaRequirementInterface. Maybe just get the required namespaces after validating the compatible version.
  *   Your latest run on php-cs-fixer was not using the default config (.php_cs.dist), this happens because the version problem and introduce some unwanted changes.

If you are interested to improve this library it would be great to work together and make a plan using something to communicate faster, I just add the gitter channel for this repo https://gitter.im/eclipxe13/buzoncfdi-cfdireader

File: tests/assets/cfdi-valid-33-sat.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
...zYxMDAzTURGUk5OMDkxETAPBgNVBAsTCFVuaWRhZCAxMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC2TTQSPONBOVxpXv9wLYo8jezBrb34i/tLx8jGdtyy27BcesOav2c1NS/Gdv10u9SkWtwdy34uRAVe7H0a3VMRLHAkvp2qMCHaZc4T8k47Jtb9wrOEh/XFS8LgT4y5OQYo6civfXXdlvxWU/gdM/e6I2lg6FGorP8H4GPAJ/qCNwIDAQABox0wGzAMBgNVHRMBAf8EAjAAMAsGA1UdDwQEAwIGwDANBgkqhkiG9w0BAQUFAAOCAQEATxMecTpMbdhSHo6KVUg4QVF4Op2IBhiMaOrtrXBdJgzGotUFcJgdBCMjtTZXSlq1S4DG1jr8p4NzQlzxsdTxaB8nSKJ4KEMgIT7E62xRUj15jI49qFz7f2uMttZLNThipunsN/NF1XtvESMTDwQFvas/Ugig6qwEfSZc0MDxMpKLEkEePmQwtZD+zXFSMVa6hmOu4M+FzGiRXbj4YJXn9Myjd8xbL/c+9UIcrYoZskxDvMxc6/6M3rNNDY3OFhBK+V/sPMzWWGt8S1yjmtPfXgFs1t65AZ2hcTwTAuHrKwDatJ1ZPfa482ZBROAAX1waz7WwXp0gso7sDCm2/yUVww==" noCertificado="20001000000100005867" formaDePago="PAGO EN UNA SOLA EXHIBICION" sello="MMakX8Q1RyedjoNEQdHN64DgpZesmkcPSye2Tb+nELqFNNzITTicyc7WRcr/gj4UQX1ffhBuWmO959453o4pkMO9omip2DiM0Ud2KitU58+P4yRyxdjhmaRS9QSsUSpcKIrLB4viSJ9GD32pJYnxmpxYi8+GO+n3HvIAMHega+A=" fecha="2015-02-18T11:49:20" folio="38" Serie="C" Version="3.3" xsi:schemaLocation="http://www.buzonfiscal.com/ns/addenda/bf/2 http://www.buzonfiscal.com/schema/xsd/Addenda_BF_v20.xsd http://www.sat.gob.mx/cfd/3 http://www.sat.gob.mx/sitio_internet/cfd/3/cfdv32.xsd ">
```
Comment (eclipxe13) [inline tests/assets/cfdi-valid-33-sat.xml:2]: This document is not 3.3, the namespace refers to `http://www.sat.gob.mx/sitio_internet/cfd/3/cfdv32.xsd`
Also, is mandatory that the namespace declaration as: `xmlns:cfdi="http://www.sat.gob.mx/cfd/3"`

File: tests/assets/cfdi-valid-33.xml [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
...zYxMDAzTURGUk5OMDkxETAPBgNVBAsTCFVuaWRhZCAxMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCBiQKBgQC2TTQSPONBOVxpXv9wLYo8jezBrb34i/tLx8jGdtyy27BcesOav2c1NS/Gdv10u9SkWtwdy34uRAVe7H0a3VMRLHAkvp2qMCHaZc4T8k47Jtb9wrOEh/XFS8LgT4y5OQYo6civfXXdlvxWU/gdM/e6I2lg6FGorP8H4GPAJ/qCNwIDAQABox0wGzAMBgNVHRMBAf8EAjAAMAsGA1UdDwQEAwIGwDANBgkqhkiG9w0BAQUFAAOCAQEATxMecTpMbdhSHo6KVUg4QVF4Op2IBhiMaOrtrXBdJgzGotUFcJgdBCMjtTZXSlq1S4DG1jr8p4NzQlzxsdTxaB8nSKJ4KEMgIT7E62xRUj15jI49qFz7f2uMttZLNThipunsN/NF1XtvESMTDwQFvas/Ugig6qwEfSZc0MDxMpKLEkEePmQwtZD+zXFSMVa6hmOu4M+FzGiRXbj4YJXn9Myjd8xbL/c+9UIcrYoZskxDvMxc6/6M3rNNDY3OFhBK+V/sPMzWWGt8S1yjmtPfXgFs1t65AZ2hcTwTAuHrKwDatJ1ZPfa482ZBROAAX1waz7WwXp0gso7sDCm2/yUVww==" noCertificado="20001000000100005867" formaDePago="PAGO EN UNA SOLA EXHIBICION" sello="MMakX8Q1RyedjoNEQdHN64DgpZesmkcPSye2Tb+nELqFNNzITTicyc7WRcr/gj4UQX1ffhBuWmO959453o4pkMO9omip2DiM0Ud2KitU58+P4yRyxdjhmaRS9QSsUSpcKIrLB4viSJ9GD32pJYnxmpxYi8+GO+n3HvIAMHega+A=" fecha="2015-02-18T11:49:20" folio="38" serie="C" version="3.3" xsi:schemaLocation="http://www.buzonfiscal.com/ns/addenda/bf/2 http://www.buzonfiscal.com/schema/xsd/Addenda_BF_v20.xsd http://www.sat.gob.mx/cfd/3 http://www.sat.gob.mx/sitio_internet/cfd/3/cfdv32.xsd ">
```
Comment (eclipxe13) [inline tests/assets/cfdi-valid-33.xml:2]: This document is not 3.3, the namespace refers to `http://www.sat.gob.mx/sitio_internet/cfd/3/cfdv32.xsd`
Also, is mandatory that the namespace declaration as: `xmlns:cfdi="http://www.sat.gob.mx/cfd/3"`

**Comment by eclipxe13** [comment]
Try to rebase to new master branch, it includes two commits:
- Fix the composer.json file
- Add information (spanish) about the namespaces

**Comment by eclipxe13** [comment]
@arturolinares please check the new release, it contains initial support for version 3.3

**Comment by arturolinares** [comment]
Acabo de actualizar, muchas gracias. Veo que hablas español también :) Veo muy bien los cambios, la documentación y TODO ofrecen una guía inicial por donde caminar.

Muchas gracias!

**Comment by eclipxe13** [comment]
De nada Arturo, en los registros de github prefiero escribir en inglés, por eso te invitaba a gitter. Ojalá te sea de utilidad esta librería, conozco un par de desarrolladores que la están utilizando y ya estaba planeando el soporte para 3.3. Aún queda mucho por hacer así que si gustas contribuir podemos platicar más abiertamente en otro canal.

PS: Voy a cerrar este pull request.

**Pull request closed without merging**